sphinx/io.py

```diff
@@ -98,6 +98,11 @@ class SphinxStandaloneReader(SphinxBaseReader):
                   RemoveTranslatableInline, PreserveTranslatableMessages, FilterSystemMessages,
                   RefOnlyBulletListTransform, UnreferencedFootnotesDetector]
 
+    def __init__(self, app, parsers={}, *args, **kwargs):
+        SphinxBaseReader.__init__(self, app, parsers, *args, **kwargs)
+        if app.env.settings['smart_quotes']:
+            self.transforms.append(SphinxSmartQuotes)
```

Comment:
Please override `get_transforms()` and add `SphinxSmartQuotes` conditionally. because `self.transforms` is a class variable. So `SphinxSmartQuotes` will be added on each instantiation.
```
def __init__(...):
    ...
    self.smart_quotes = app.env.settings['smart_quotes']

def get_transforms(self):
    transforms = SphinxBaseReder.get_transforms()
    if self.smart_quotes:
        transforms.append(SphinxSmartQuotes)

    return transforms
```

Reply:
sorry for my basic mistake. Indeed build time considerably increased due to it. I will fix.